x/axelarcork/keeper/keeper.go

```diff
@@ -0,0 +1,399 @@
+func NewKeeper(
+	cdc codec.BinaryCodec, key sdk.StoreKey, paramSpace paramtypes.Subspace,
+	stakingKeeper types.StakingKeeper, transferKeeper types.TransferKeeper,
+	distributionKeeper types.DistributionKeeper, wrapper types.ICS4Wrapper,
+) Keeper {
+	// set KeyTable if it has not already been set
+	if !paramSpace.HasKeyTable() {
+		paramSpace = paramSpace.WithKeyTable(types.ParamKeyTable())
+	}
+
+	return Keeper{
+		storeKey:           key,
+		cdc:                cdc,
+		paramSpace:         paramSpace,
+		stakingKeeper:      stakingKeeper,
+		transferKeeper:     transferKeeper,
+		distributionKeeper: distributionKeeper,
+
+		ics4Wrapper: wrapper,
+	}
+}
+
+// Logger returns a module-specific logger.
+func (k Keeper) Logger(ctx sdk.Context) log.Logger {
+	return ctx.Logger().With("module", "x/"+types.ModuleName)
+}
+
+////////////
+// Params //
+////////////
+
+// GetParamSet returns the vote period from the parameters
+func (k Keeper) GetParamSet(ctx sdk.Context) types.Params {
+	var p types.Params
+	k.paramSpace.GetParamSet(ctx, &p)
+	return p
+}
+
+// setParams sets the parameters in the store
+func (k Keeper) setParams(ctx sdk.Context, params types.Params) {
+	k.paramSpace.SetParamSet(ctx, &params)
+}
+
+/////////////////////
+// Scheduled Corks //
+/////////////////////
+
+func (k Keeper) SetScheduledCork(ctx sdk.Context, chainID uint64, blockHeight uint64, val sdk.ValAddress, cork types.Cork) []byte {
+	id := cork.IDHash(blockHeight)
+	bz := k.cdc.MustMarshal(&cork)
+	ctx.KVStore(k.storeKey).Set(types.GetScheduledCorkKey(chainID, blockHeight, id, val, common.HexToAddress(cork.TargetContractAddress)), bz)
+	return id
+}
+
+// GetScheduledCork gets the scheduled cork for a given validator
+// CONTRACT: must provide the validator address here not the delegate address
+func (k Keeper) GetScheduledCork(ctx sdk.Context, chainID uint64, blockHeight uint64, id []byte, val sdk.ValAddress, contract common.Address) (types.Cork, bool) {
+	store := ctx.KVStore(k.storeKey)
+
+	bz := store.Get(types.GetScheduledCorkKey(chainID, blockHeight, id, val, contract))
+	if len(bz) == 0 {
+		return types.Cork{}, false
+	}
+
+	var cork types.Cork
+	k.cdc.MustUnmarshal(bz, &cork)
+	return cork, true
+}
+
+// DeleteScheduledCork deletes the scheduled cork for a given validator
+// CONTRACT: must provide the validator address here not the delegate address
+func (k Keeper) DeleteScheduledCork(ctx sdk.Context, chainID uint64, blockHeight uint64, id []byte, val sdk.ValAddress, contract common.Address) {
+	ctx.KVStore(k.storeKey).Delete(types.GetScheduledCorkKey(chainID, blockHeight, id, val, contract))
+}
+
+// IterateScheduledCorks iterates over all scheduled corks in the store
+func (k Keeper) IterateScheduledCorks(ctx sdk.Context, chainID uint64, cb func(val sdk.ValAddress, blockHeight uint64, id []byte, cel common.Address, cork types.Cork) (stop bool)) {
+	k.IterateScheduledCorksByPrefix(ctx, types.GetScheduledCorkKeyPrefix(chainID), cb)
+}
```

Comment:
I think this can be discussed. Since there is (as far as I can tell) no reason to iterate all corks across all chains at once, and since all actions in the Axelar Cork module either take a chain ID, or modify chain configurations, is that specification necessary?

Reply:
Not a strong opinion, the phrase "in the store" just seems pretty specifically wrong imo

Reply:
Ah gotcha. Updated